relay-server/src/utils/rate_limits.rs

```diff
@@ -474,7 +496,7 @@ where
     ///    are exhausted, both the event and metrics are dropped.
     ///  - `DataCategory::TransactionIndexed` counts ingested and stored events. If quotas with this
     ///    category are exhausted, just the event payload is dropped, but metrics are kept.
-    fn index_category(&self, category: DataCategory) -> Option<DataCategory> {
+    fn transaction_index_category(&self, category: DataCategory) -> Option<DataCategory> {
```

Comment:
This function explicitly checks for `DataType::Transaction`, so I renamed it to make its purpose more clear.